src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```
@@ -176,6 +179,18 @@ private static void WarnForHttpSources(ListPackageArgs listPackageArgs, ListPack
                 }
             }
 
+            foreach (PackageSource packageSource in listPackageArgs.AuditSources)
+            {
+                if (packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections)
```

There's no `else` block, so if any sources defined in nuget.config are not used because it's using insecure http, it will be silently ignored. I believe all or other commands report an error, so this should as well.

---

There is a logic that follows this block of code that does the logging. This is just adding to the list of http package sources so that they can be logged. I am not sure I understand why we need an else block